materialize-bigquery/bigquery.go

```diff
@@ -21,39 +22,30 @@ func main() {
 	boilerplate.RunMain(newBigQueryDriver())
 }
 
-// Config represents the endpoint configuration for BigQuery.
-type config struct {
-	BillingProjectID string `json:"billing_project_id,omitempty"`
-	ProjectID        string `json:"project_id"`
-	Dataset          string `json:"dataset"`
-	Region           string `json:"region,omitempty"`
-	Bucket           string `json:"bucket"`
-	BucketPath       string `json:"bucket_path"`
-	CredentialsFile  string `json:"credentials_file,omitempty"`
-	CredentialsJSON  string `json:"credentials_json,omitempty"`
-}
+type credential string
 
-func (*config) GetFieldDocString(fieldname string) string {
-	switch fieldname {
-	case "BillingProjectID":
-		return "Billing Project ID connected to the BigQuery dataset. It can be the same value as Project ID."
-	case "ProjectID":
-		return "Google Cloud Project ID that owns the BigQuery dataset."
-	case "Dataset":
-		return "BigQuery dataset that will be used to store the materialization output."
-	case "Region":
-		return "Region where both the Bucket and the BigQuery dataset is located. They both need to be within the same region."
-	case "Bucket":
-		return "Google Cloud Storage bucket that is going to be used to store specfications & temporary data before merging into BigQuery."
-	case "BucketPath":
-		return "A prefix that will be used to store objects to Google Cloud Storage's bucket."
-	case "CredentialsFile":
-		return "URI that points to a JSON Credentials for Google Clouse Service."
-	case "CredentialsJSON":
-		return "Google Cloud Service Account JSON credentials in base64 format."
-	default:
-		return ""
+func (credential) JSONSchemaType() *jsonschema.Type {
```

Comment:
This seems like a pretty reasonable pattern for customizing schema generation, given that the go struct tags won't work :+1: